geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/serial/SerialWANStatsDUnitTest.java

```diff
@@ -86,6 +86,265 @@ public void testReplicatedSerialPropagation() throws Exception {
+
+    vm5.invoke(() -> WANTestBase.checkQueueStats("ln", 0, entries, 0, 0));
+    vm5.invoke(() -> WANTestBase.checkBatchStats("ln", 0, true));
+  }
+
+  @Test
+  public void testReplicatedSerialPropagationTransactionBatchRedistributedNoGroupTransactions()
+      throws Exception {
+    Integer lnPort = (Integer) vm0.invoke(() -> WANTestBase.createFirstLocatorWithDSId(1));
+    Integer nyPort = (Integer) vm1.invoke(() -> WANTestBase.createFirstRemoteLocator(2, lnPort));
+
+    createCacheInVMs(nyPort, vm2);
+
+    createCacheInVMs(lnPort, vm4, vm5, vm6, vm7);
+
+    int batchTimeInterval = 30000;
+    boolean isGroupTransactionEvents = false;
+    vm4.invoke(
+        () -> WANTestBase.createSender("ln", 2, false, 100, 10, false, false, null, true,
+            isGroupTransactionEvents,
+            batchTimeInterval));
+    vm5.invoke(
+        () -> WANTestBase.createSender("ln", 2, false, 100, 10, false, false, null, true,
+            isGroupTransactionEvents,
+            batchTimeInterval));
+    vm6.invoke(
+        () -> WANTestBase.createSender("ln", 2, false, 100, 10, false, false, null, true,
+            isGroupTransactionEvents,
+            batchTimeInterval));
+    vm7.invoke(
+        () -> WANTestBase.createSender("ln", 2, false, 100, 10, false, false, null, true,
+            isGroupTransactionEvents,
+            batchTimeInterval));
+
+    vm2.invoke(() -> WANTestBase.createReplicatedRegion(testName + "_RR", null, isOffHeap()));
+
+    startSenderInVMs("ln", vm4, vm5);
+
+    vm4.invoke(() -> WANTestBase.createReplicatedRegion(testName + "_RR", "ln", isOffHeap()));
+    vm5.invoke(() -> WANTestBase.createReplicatedRegion(testName + "_RR", "ln", isOffHeap()));
+    vm6.invoke(() -> WANTestBase.createReplicatedRegion(testName + "_RR", "ln", isOffHeap()));
+    vm7.invoke(() -> WANTestBase.createReplicatedRegion(testName + "_RR", "ln", isOffHeap()));
+
+    final Map keyValues = new HashMap();
+    int entries = 24;
+    for (int i = 0; i < entries; i++) {
+      keyValues.put(i, i + "_Value");
+    }
+    int eventsPerTransaction = 3;
+    vm5.invoke(() -> WANTestBase.doPutsInsideTransactions(testName + "_RR", keyValues,
+        eventsPerTransaction));
+
+    // The receiver is started later in order for the batch to be redistributed (sent again)
+    Thread.sleep(2000);
+    vm2.invoke(() -> WANTestBase.createReceiver());
+
+    vm2.invoke(() -> WANTestBase.validateRegionSize(testName + "_RR", entries));
+
+    pause(2000);
+    vm2.invoke(() -> WANTestBase.checkGatewayReceiverStats(3, entries, entries, true));
+
+    vm4.invoke(() -> WANTestBase.checkQueueStats("ln", 0, entries, entries, entries));
+    vm4.invoke(() -> WANTestBase.checkBatchStats("ln", 3, true, true));
+
+    vm5.invoke(() -> WANTestBase.checkQueueStats("ln", 0, entries, 0, 0));
+    vm5.invoke(() -> WANTestBase.checkBatchStats("ln", 0));
+  }
+
+  @Test
+  public void testReplicatedSerialPropagationTransactionBatchRedistributedGroupTransactions()
```

Comment:
This test name could be more descriptive.

Reply:
How about `tesReplicatedSerialPropagationWithBatchRedistWithGroupTransactionEventsSendsBatchesWithoutIncompleteTransactions`?

Reply:
Sounds good